snakemake/dag.py

```diff
@@ -69,15 +69,18 @@ def get_batch(self, items: list):
                 "Please choose a smaller number of batches.".format(self.rulename)
             )
         items = sorted(items)
-        batch_len = math.floor(len(items) / self.batches)
+        k, m = divmod(len(items), self.batches)
+
         # self.batch is one-based, hence we have to subtract 1
         idx = self.idx - 1
-        i = idx * batch_len
+        i = idx * k + min(idx, m)
+        batch_len = (idx + 1) * k + min(idx + 1, m)
```

Comment:
May I ask you to add some comments explaining each line for the future?

Reply:
Good point, I just pushed a new version with comments.